draft-ietf-anima-constrained-join-proxy.md

```diff
@@ -316,30 +316,32 @@ The advantages and disadvantages of the two modes are presented in {{jp-comparis
 For a Join Proxy implementation on a node, there are three possible scenarios:
 
 1. Both stateful and stateless modes are implemented. The Join Proxy can switch between these modes, depending on 
-   configuration.
+   configuration and/or auto-discovery of registrar(s) for each option.
 2. Only stateful mode is implemented. 
 3. Only stateless mode is implemented.
 
-An application profile or ecosystem standard that integrates the Join Proxy functionality as defined in this 
-document MAY define any of these three options. 
-In particular, option 2 or 3 has the advantage of reducing code size and testing efforts, when all devices under 
-the application profile/standard adhere to the same choice.
+Option 2 and 3 have the advantage of reducing code size, testing efforts and deployment complexity,
+but requires for all devices in the deployment to standardize on the same choice.
 
-A generic Join Proxy that is not adhering to such an application profile/standard MUST implement both modes. 
+An standard for a network-wide application or ecosystem profile integrates the Join Proxy functionality
+as defined in this document MAY specify the use of any of these three options. It is expected that most
+deployments of constrained Join Proxies will be in the context of such standards and that they will be able
+to pick either 2 or 3 based on considerations such as those from {{jp-comparison}}.
 
-A cBRSKI Registrar by design necessarily implements the stateful mode, and it SHOULD implement support for 
-Join Proxies operating in the stateless mode. The exception case here is a cBRSKI Registrar that is implemented for a 
-particular dedicated application profile/standard which specifies only the stateful mode.
+A Join Proxy that is not adhering to such an additionals standard MUST implement both modes. 
+A Join Proxy or Registrar not complying to such additional standards are called "generic".
 
-If a Join Proxy implements both modes, then it MUST use only the mode that is currently configured for the network 
-(by a method or profile outside the scope of this document) or the mode individually configured for the device.
-If the mode is not configured, the device MUST NOT operate as a Join Proxy.
+If a Join Proxy implements both modes and does not implement methods to discover available registrar
```

Comment:
First sentence here to be discussed - not clear what it intends to say ("for either method which are outside the scope of this document" ?)   A bit too much combined in one sentence, I think.